core/frontend/src/types/autopilot/parameter-table.ts

```diff
@@ -94,29 +97,45 @@ export default class ParametersTable {
     this.parametersDict = {}
   }
 
+  async fetchArduPilotMetadata(): Promise<MetadataFile | string> {
+    await fetchFirmwareVehicleType() // required to populate autopilot.vehicle_type
+    const jsons = Object.keys(await import.meta.glob('/public/assets/ArduPilot-Parameter-Repository/**/*.json')) as string[]
+    let folder = "Copter"
+    switch (autopilot.firmware_vehicle_type) {
+      case FirmwareVehicleType.ArduSub:
+        folder = 'Sub'
+        break
+      case FirmwareVehicleType.ArduRover:
+        folder = 'Rover'
+        break
+      case FirmwareVehicleType.ArduPlane:
+        folder = 'Plane'
+    }
+    const major = autopilot.firmware_info?.version.major
+    let minor = autopilot.firmware_info?.version.minor ?? 0
+    while (minor >= 0) {
+      // go down one minor version at a time until we find a match
+      const metadata_path = `/assets/ArduPilot-Parameter-Repository/${folder}-${major}.${minor}/apm.pdef.json`
+      if (jsons.includes(`/public${metadata_path}`)) {
+        return await axios.get(metadata_path).then(response => response.data as MetadataFile)
+      }
+      minor -= 1;
+    }
+    // if we didn't find a match, just return the first one. that must be better than nothing, right?
+    const fallback = jsons[0].replace('/public', '')
+    console.warn(`Could not find metadata for ${folder}-${major}.X. Falling back to ${fallback}`)
+    return axios.get(fallback).then(response => response.data as MetadataFile)
+  }
+
+
   async fetchMetadata(): Promise<void> {
     if (autopilot.vehicle_type === null) {
       // Check again later if we have a vehicle type identified
       fetchVehicleType()
       setTimeout(() => { this.fetchMetadata() }, 1000)
       return
     }
-
-    if (autopilot_data.autopilot_type === MavAutopilot.MAV_AUTOPILOT_PX4) {
-      this.metadata = fromPX4toArduPilotParametersMetadata(await fetchPX4Metadata())
```

Comment:
I think that the PX4 case was removed by mistake

Reply:
oh no, yeah, definitely a mistake :p